### PR TITLE
Update AWSCore CDK to CDKv2

### PR DESCRIPTION
**Pull request by lmbr-pip**
Signed-off-by: Pip Potter <61438964+lmbr-pip@users.noreply.github.com>

## What does this PR do?

Updates the AWSCore CDK to CDKv2. Builds off https://github.com/o3de/o3de/pull/10120

## How was this PR tested?

```
cdk synth
cdk deploy (both stacks)
python example/dynamodb_table_seeder.py --table_name O3DE-AWS-PROJECT-AWSCore-Example-us-east-1-O3DEAWSPROJECTAWSCoreTable92747B57-6KLWLJC9ROAB --region us-east-1
```

### CODE REVIEW

File: Gems/AWSCore/cdk/README.md [hunk 1]
```diff
@@ -11,14 +15,21 @@ The `cdk.json` file tells the CDK Toolkit how to execute your app.
...
+(or `python` for Windows) (Python 3.7+) executable in your path with access to the `venv`
 package. If for any reason the automatic creation of the virtualenv fails,
 you can create the virtualenv manually.
 
-See https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html about for information about how to set up
+See https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html about for information about how to set up
 the prerequisites for CDK development.
 
-To manually create a virtualenv on MacOS and Linux:
+See for more information about working with the CDK and O3DE, see the [Deploying the CDK Application](https://www.o3de.org/docs/user-guide/gems/reference/aws/aws-core/cdk-application/)
```
Comment (allisaurus) [inline Gems/AWSCore/cdk/README.md:25]: can remove "See..." at beginning, so it read like:
"For more information about working with the CDK and O3DE, see the `Deploying the CDK Application` section..."

File: Gems/AWSCore/cdk/README.md [hunk 2]
```diff
@@ -11,14 +15,21 @@ The `cdk.json` file tells the CDK Toolkit how to execute your app.
...
+section on o3de.org. 
+
+> Note: This stack is for CDK v2 (the latest CDK version, if you are working with CDKv1 stacks please use the CDKv1 version of this application).
```
Comment (allisaurus) [inline Gems/AWSCore/cdk/README.md:28]: nit: it feels like the parenthetical should close before the comma, so it reads like:
"Note: This stack is for CDK v2 (the latest CDK version), if you are working with CDKv1 stacks please use the CDKv1 version of this application."

File: Gems/AWSCore/cdk/README.md [hunk 3]
```diff
@@ -58,11 +69,9 @@ At this point you can now synthesize the CloudFormation template for this code.
 $ cdk synth
 ```
 
-You may need todo a one time bootstrap, once per account, per region. The CDK application will prompt you on this.
+You may need todo a one time [bootstrap](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html), once per account, per region. The CDK application will prompt you on this.
```
Comment (allisaurus) [inline Gems/AWSCore/cdk/README.md:72]: need space between "to/do"

**Review by allisaurus — APPROVED** [review]
LGTM, just a couple minor wording/grammar notes. Thanks for upgrading this to v2!

File: Gems/AWSCore/cdk/README.md (hunk 2 above)
Comment (lmbr-pip) [reply Gems/AWSCore/cdk/README.md:28]: Fixed.

File: Gems/AWSCore/cdk/README.md (hunk 1 above)
Comment (lmbr-pip) [reply Gems/AWSCore/cdk/README.md:25]: Fixed

**Pull request merged**